src/Fixer/Operator/IncrementStyleFixer.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Operator;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Tokenizer\TokensAnalyzer;
+
+/**
+ * @author Gregor Harlan <gharlan@web.de>
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+final class IncrementStyleFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Pre or post increment and decrement operators should be used if possible.',
+            [new CodeSample("<?php\n\$a++;\n\$b--;")]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound([T_INC, T_DEC]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('style', 'Whether to use `pre` or `post` increment and decrement operators.'))
+                ->setAllowedValues(['pre', 'post'])
+                ->setDefault('pre')
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        for ($index = $tokens->count() - 1; 0 <= $index; --$index) {
+            $token = $tokens[$index];
+
+            if (!$token->isGivenKind([T_INC, T_DEC])) {
+                continue;
+            }
+
+            if ('post' === $this->configuration['style'] && $tokensAnalyzer->isUnaryPredecessorOperator($index)) {
+                $prevToken = $tokens[$tokens->getPrevMeaningfulToken($index)];
+                if (!$prevToken->equalsAny([';', '{', '}', [T_OPEN_TAG]])) {
+                    continue;
+                }
+
+                $endIndex = $this->findEnd($tokens, $index);
+
+                $nextToken = $tokens[$tokens->getNextMeaningfulToken($endIndex)];
+                if ($nextToken->equalsAny([';', ')'])) {
+                    $tokens->clearAt($index);
+                    $tokens->insertAt($tokens->getNextNonWhitespace($endIndex), clone $token);
+                }
+            } elseif ('pre' === $this->configuration['style'] && $tokensAnalyzer->isUnarySuccessorOperator($index)) {
```

Comment:
const for `pre` is missing

Reply:
do you mean to extract `pre` and `post` to class constants?

Reply:
yes